src/mlpack/bindings/R/mlpack/inst/CITATION

```diff
@@ -0,0 +1,49 @@
+bibentry("Manual",
+         other = unlist(citation(auto = meta), recursive = FALSE))
```

Comment:
Nice trick! I've not seen that one before.

Looks like the delayed promise execution causes the `meta` object to be automatically defined when `citation()` is called? 

```r
if (!is.null(auto) && !is.logical(auto) && !anyNA(match(c("Package", 
        "Version", "Title"), names(meta <- as.list(auto)))) && 
        !all(is.na(match(c("Authors@R", "Author"), names(meta))))) {
        auto_was_meta <- TRUE
        package <- meta$Package
    }
```

Reply:
I _literally_ stole^Hborrowed this whole first entry for 'the package' from one of Kurt's own packages and its CITATION file, site unseen. Worked when I tried it and carried it over to two or three my packages.

How to automagically get the package name within the scope of a file in the package had bugged me for some time.   I still do it 'explicitly' in the testrunner invocation in `tests/*R` which is generally tinytest for me.